src/rascal/structure_managers/structure_manager.hh

```diff
@@ -400,88 +447,162 @@ namespace rascal {
+      error << "No property of name '" << name << "' has been registered";
+      throw std::runtime_error(error.str());
     }
 
-    template <typename T, size_t Order, Dim_t NbRow = 1, Dim_t NbCol = 1>
-    std::shared_ptr<Property_t<T, Order, NbRow, NbCol>>
-    get_property(const std::string & name) const {
-      return this
-          ->template get_validated_property<Property_t<T, Order, NbRow, NbCol>>(
-              name);
+    /**
+     * For MD when properties needs to be updated
+     */
+    template <bool IsRoot = IsRootImplementation,
+              std::enable_if_t<IsRoot, int> = 0>
+    inline void set_updated_property_status(const bool is_updated) {
+      for (auto & element : this->properties) {
+        auto & property{element.second};
+        property->set_updated_status(is_updated);
+      }
     }
 
-    void set_updated_property_status(bool is_updated) {
+    template <bool IsRoot = IsRootImplementation,
+              std::enable_if_t<not(IsRoot), int> = 0>
+    inline void set_updated_property_status(const bool is_updated) {
       for (auto & element : this->properties) {
         auto & property{element.second};
         property->set_updated_status(is_updated);
       }
+      return this->get_previous_manager()->set_updated_property_status(
+          is_updated);
     }
 
     void set_updated_property_status(const std::string & name,
                                      bool is_updated) {
-      this->properties[name]->set_updated_status(is_updated);
+      if (this->has_layer_property(name)) {
+        this->properties[name]->set_updated_status(is_updated);
+        return;
+      }
+      std::stringstream error{};
+      error << "A property of name '" << name << "' does not exist"
+            << " in manager '" << this->get_name() << "' on top layer";
+      throw std::runtime_error(error.str());
+    }
+
+    /*
+     * Forwards property requersts to lower layers. Usually to get a property
+     * the `get_propertp_ptr` or `get_property_ref` function should be used.
+     * This function is however still public because of each structure manager
+     * needs to be able to access it from the previous manager.
+     */
+    template <typename UserProperty_t, bool IsRoot = IsRootImplementation,
+              std::enable_if_t<IsRoot, int> = 0>
+    std::shared_ptr<UserProperty_t>
+    forward_get_property_request(const std::string & name,
+                                 const bool validate_property) {
+      if (this->has_layer_property(name)) {
+        auto property = this->properties.at(name);
+        if (validate_property) {
+          this->template validate_property_t<UserProperty_t>(property);
+        }
+        return std::static_pointer_cast<UserProperty_t>(property);
+      }
+      std::stringstream error{};
+      error << "A property of name '" << name << "' does not exist"
+            << " in manager '" << this->get_name() << "'";
+      throw std::runtime_error(error.str());
```

Comment:
is this function meant to be used by user ? if not please move it to the protected section

Reply:
it is not meant to be used by the user, but by other adaptors, therefore it needs to be public
AdaptorNL< StructureManagerCenter >
AdatporNL invokes `forward_get_property_request` from StructureManagerCenter https://github.com/cosmo-epfl/librascal/blob/d8dc2fe43a2cee5d276e940dfa041cecd9e1072b/src/rascal/structure_managers/structure_manager.hh#L583-L585